metadata/metadata_test.go

```diff
@@ -0,0 +1,326 @@
+					Annotations: map[string]string{"pet/dog": "ruff"},
+				},
+			}
+			AddAnnotations(pipelineRun, release.GetAnnotations())
+			It("should copy all src key/value pairs into dst that do not already exist", func() {
+				Expect(pipelineRun.GetAnnotations()["pet/dog"]).To(Equal("ruff"))
+				Expect(pipelineRun.GetAnnotations()["pet/cat"]).To(Equal("meow"))
+				Expect(pipelineRun.GetAnnotations()["pond/frog"]).To(Equal("ribit"))
+			})
+			It("should have increased the length of the dst map", func() {
+				Expect(len(pipelineRun.GetAnnotations())).To(Equal(3))
+			})
+		})
+		When("called with a pipelineRun containing nil Annotations", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels:      map[string]string{},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: nil,
+					Labels:      nil,
+				},
+			}
+			AddAnnotations(pipelineRun, release.GetAnnotations())
+			It("should create the Annotations map", func() {
+				Expect(pipelineRun.GetAnnotations()).ToNot(BeNil())
+			})
+		})
+	})
+
+	Context("metadata.AddLabels method", func() {
+		When("calls the addEntries function with an empty map", func() {
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"pet/dog": "ruff"},
+				},
+			}
+			AddLabels(pipelineRun, map[string]string{})
+			It("src and dst maps should have the same length", func() {
+				Expect(len(pipelineRun.GetLabels())).To(Equal(1))
+			})
+		})
+		When("calls the addEntries function with a non-empty map", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels: map[string]string{
+						"pet/dog":   "bark",
+						"pet/cat":   "meow",
+						"pond/frog": "ribit",
+					},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"pet/dog": "ruff"},
+				},
+			}
+			AddLabels(pipelineRun, release.GetLabels())
+			It("should copy all src key/value pairs into dst that do not already exist", func() {
+				Expect(pipelineRun.GetLabels()["pet/dog"]).To(Equal("ruff"))
+				Expect(pipelineRun.GetLabels()["pet/cat"]).To(Equal("meow"))
+				Expect(pipelineRun.GetLabels()["pond/frog"]).To(Equal("ribit"))
+			})
+			It("should have increased the length of the dst map", func() {
+				Expect(len(pipelineRun.GetLabels())).To(Equal(3))
+			})
+		})
+		When("called with a pipelineRun containing nil Labels", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels:      map[string]string{},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: nil,
+					Labels:      nil,
+				},
+			}
+			AddLabels(pipelineRun, release.GetAnnotations())
+			It("should create the Annotations map", func() {
```

Comment:
Good catch.
Turns out the `release` object wasn't really needed in that test anymore, so I've updated this to the concrete type `map[string]string`

Reply:
This line should still have Annotations replaced with Labels

Reply:
Thanks!
I apparently missed this one in the most recent refactor, but actually mistakenly _**thought**_ I had fixed it. To use a cliche, it got swept under the rug... So my apologies if I resolved this one by mistake. :disappointed: 
I appreciate your relentless attention to details like this. :+1: 

FYI - instead of stacking another commit, I'm simply force pushing over top, and clobbering the previous. So this may show as "outdated".